pysatSpaceWeather/instruments/methods/ace.py

```diff
@@ -56,13 +61,232 @@ def references(iname):
+
+    if inst.platform != "ace":
+        raise AttributeError("Can't apply ACE cleaning to platform {:}".format(
+            inst.platform))
+
+    # Set the clean level
+    if inst.clean_level == 'dusty':
+        logger.warning("unused clean level 'dusty', reverting to 'clean'")
+        inst.clean_level = 'clean'
+
+    max_status = 9
+    if inst.clean_level == "clean":
+        max_status = 0
+    elif inst.clean_level == "dirty":
+        max_status = 8
+
+    # Replace all fill values with NaN
+    for col in inst.data.columns:
+        fill_val = inst.meta[col][inst.meta.labels.fill_val]
+        if ~np.isnan(fill_val):
+            inst.data[col] = inst.data[col].replace(fill_val, np.nan)
+
+    return max_status
+
+
+def list_files(name='', tag='', inst_id='', data_path='', format_str=None):
+    """Return a Pandas Series of every file for ACE data
+
+    Parameters
+    ----------
+    name : str
+        ACE Instrument name. (default='')
+    tag : str
+        Denotes type of file to load. (default='')
+    inst_id : str
+        Specifies the ACE instrument ID. (default='')
+    data_path : str
+        Path to data directory. (default='')
+    format_str : string or NoneType
+        User specified file format.  If None is specified, the default
+        formats associated with the supplied tags are used. (default=None)
+
+    Returns
+    -------
+    pysat.Files.from_os : pysat.utils.files.Files
+        A class containing the verified available files
+
+    Note
+    ----
+    Called by pysat. Not intended for direct use by user.
+
+    """
+    if format_str is None:
+        format_str = '_'.join(["ace", name, tag,
+                               '{year:04d}-{month:02d}-{day:02d}.txt'])
+
+    files = pysat.Files.from_os(data_path=data_path, format_str=format_str)
+
+    return files
+
+
+def download(date_array, name='', tag='', inst_id='', data_path='',
+             now=dt.datetime.utcnow()):
+    """Routine to download ACE Space Weather data
+
+    Parameters
+    ----------
+    date_array : array-like
+        Array of datetime values
+    name : str
+        ACE Instrument name. (default='')
+    tag : str
+        Denotes type of file to load. Accepted types are 'realtime' and
+        'historic'. (default='')
+    inst_id : str
+        Specifies the ACE instrument ID. (default='')
+    data_path : str
+        Path to data directory. (default='')
+    now : dt.datetime
+        Current universal time (default=dt.datetime.utcnow())
+
+    Note
+    ----
+    Called by pysat. Not intended for direct use by user.
+
+    Warnings
+    --------
+    Only able to download current real-time data
+
+    """
+    # Define the file information for each data type and check the
+    # date range
+    if tag == 'realtime':
+        file_fmt = "{:s}-{:s}.txt".format("ace", "magnetometer"
+                                          if name == "mag" else name)
+
+        if(len(date_array) > 1 or date_array[0].year != now.year
```

Comment:
What if someones starts up pysat one day, and then goes to download data a few days down the line? 
I think `now` will be out of date since function default arguments are only run once when the function is defined.

Could be a place for a default of `None` and then define `now` within.